WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductVariationFormViewModel.swift

```diff
@@ -228,7 +228,18 @@ extension ProductVariationFormViewModel {
                              dateOnSaleEnd: Date?,
                              taxStatus: ProductTaxStatus,
                              taxClass: TaxClass?) {
-        let subscription = productVariation.subscription?.copy(period: subscriptionPeriod,
+        // Sets "Expire after" to "0" (i.e. Never expire)
+        // if the subscription period or interval is changed
+        let subscriptionLength: String? = {
+            if productVariation.subscription?.period != subscriptionPeriod || productVariation.subscription?.periodInterval != subscriptionPeriodInterval {
+                return "0"
+            } else {
```

Comment:
Just wondering: I cannot get my head around this logic even with the comment above. @selanthiraiyan could you please explain again why we should set the length to 0 in this case?

Reply:
- When the period is changed, the expiry date options (in drop down) will change as well. 
- To avoid selecting a non-matching expiry date value we are resetting the expiry date back to "Never expire".
- The user will have to select a valid expiry date. The expiry date options will be calculated based on the entered period. 

Here is the web behaviour. 

https://github.com/woocommerce/woocommerce-ios/assets/524475/db8c5aee-187d-4444-a1b0-2e9c9ef53d47

Here is the discussion in which @JorgeMucientes shared about this behaviour - p1700124102155059/1700121516.912349-slack-C03L1NF1EA3

Kindly let me know if you have any questions.

Reply:
Oh, this makes sense! It was hard to understand because the two fields are on separate screens, making it hard to connect them.